src/terrain/globe_raster_program.js

```diff
@@ -78,15 +92,29 @@ const globeRasterUniformValues = (
     globeMercatorMatrix: Float32Array,
     zoomTransition: number,
     mercCenter: [number, number],
-    gridMatrix: ?Array<number>
+    gridMatrix: ?Array<number>,
+    frustumDirTl: ?[number, number, number],
+    frustumDirTr: ?[number, number, number],
+    frustumDirBr: ?[number, number, number],
+    frustumDirBl: ?[number, number, number],
+    globePosition: ?[number, number, number],
+    globeRadius: ?number,
+    viewport: ?[number, number]
 ): UniformValues<GlobeRasterUniformsType> => ({
     'u_proj_matrix': Float32Array.from(projMatrix),
     'u_globe_matrix': globeMatrix,
     'u_merc_matrix': globeMercatorMatrix,
     'u_zoom_transition': zoomTransition,
     'u_merc_center': mercCenter,
     'u_image0': 0,
-    'u_grid_matrix': gridMatrix ? Float32Array.from(gridMatrix) : new Float32Array(9)
+    'u_grid_matrix': gridMatrix ? Float32Array.from(gridMatrix) : new Float32Array(9),
+    'u_frustum_tl': frustumDirTl ? frustumDirTl : [0, 0, 0],
+    'u_frustum_tr': frustumDirTr ? frustumDirTr : [0, 0, 0],
+    'u_frustum_br': frustumDirBr ? frustumDirBr : [0, 0, 0],
+    'u_frustum_bl': frustumDirBl ? frustumDirBl : [0, 0, 0],
+    'u_globe_pos': globePosition ? globePosition : [0, 0, 0],
+    'u_globe_radius': globeRadius ? globeRadius : 0,
+    'u_viewport': viewport ? viewport : [0, 0],
```

Comment:
We'll be able to mark these as non-optional when adding that to the globe poles, refer https://github.com/mapbox/mapbox-gl-js/commit/836fd715dd2b61d11f8b74bfc8c2714c866f5c99#diff-3792b2c1cdbd4785ab8f011633f43316933901d319dc7b14b664eb091a464660R111 as a reference commit.